### DISCUSSION TITLE
Http/2 - Mercure Protocol

### DISCUSSION THREAD

**Discussion by Adrian-SolM** [Azure/static-web-apps#832] [Ideas]
The first question, which I can't seem to figure out is "Does Azure Static Web apps utilize http/2 in any form?"

With the assumption of it being yes, would it be possible to host something like the [Mercure ](https://mercure.rocks/) Hub in a function, allowing for a web socket like connection. It seems like being able to have a static web app that has live updates via mercure is a possibility, and even may look to be possible to host in an azure function. The only thing being that Http/2 support would be the most ideal scenario, as well as the ability to utilize Server Sent Events (sse). 

What are your thoughts or ideas on an implementation utilizing something like this?

**Comment by sig9** [comment]
Not sure about http/2. As for streaming with node via serverless functions in azure, it is not currently possible. It is possible in c#. 
https://github.com/Azure/azure-functions-host/issues/1361

My understanding is that in the serverless runtime c# runs in process and has a stream response option. All other languages run out of process and stream responses are not supported. 

I've read that there is work to move c#7 functions to out of process. If the ablity to stream is preserved in the new function runtimes it may eventually arrive in node & other languages.  I have not seen a clear presentation of the roadmap for node.

Looking forward to any good news in this area. Azure static web apps have a ton of great features otherwise.
